### PR TITLE
[ticket/15662] Add $this->template to core.modify_notification_message

### PR DESCRIPTION
**Pull request by senky**
PHPBB3-15662

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15662

### CODE REVIEW

**Comment by senky** [comment]
!set Event

File: phpBB/includes/functions_messenger.php [hunk 1]
```diff
@@ -336,13 +337,16 @@ function send($method = NOTIFY_EMAIL, $break = false)
 		*						and the message without sending it
 		* @var	string	subject	The message subject
 		* @var	string	message	The message text
+		* @var	\phpbb\template\template	template	Template object
 		* @since 3.1.11-RC1
+		* @changed 3.2.3 Added template
```
Comment (Nicofuma) [inline phpBB/includes/functions_messenger.php]: 3.2.4 please

**Comment by rubencm** [comment]
Why?, they can get it with di

**Comment by senky** [comment]
No, this template is different one than the one from the container.

**Comment by rubencm** [comment]
Uhm, yes, but it looks the same but instanced manually

**Comment by csnover** [comment]
FWIW I think this functionality should be in a different hook. AFAICT, the `core.modify_notification_message` hook is just plain broken right now, since the very next line after the hook is blowing away `$this->msg` (and sometimes `$this->subject`) with the result of the template. Modifying the template cannot be done from the same hook that modifies the message since the template has to run before there is a message body to modify.

**Comment by csnover** [comment]
I opened [PHPBB3-15761](https://tracker.phpbb.com/browse/PHPBB3-15761) for the hook bug. Hopefully I filled out everything correctly.

File: phpBB/includes/functions_messenger.php [hunk 2]
```diff
@@ -336,13 +337,16 @@ function send($method = NOTIFY_EMAIL, $break = false)
 		*						and the message without sending it
 		* @var	string	subject	The message subject
 		* @var	string	message	The message text
+		* @var	\phpbb\template\template	template	Template object
 		* @since 3.1.11-RC1
+		* @changed 3.2.4 Added template
```
Comment (marc1706) [inline phpBB/includes/functions_messenger.php]: 3.2.4-RC1

**Comment by marc1706** [comment]
@csnover You're correct in that there are issues with modifying the subject and message. Passing the template class should however help with reducing this issue.

**Pull request merged**